staker/staker.go

```diff
@@ -403,6 +542,17 @@ func (s *Staker) Start(ctxIn context.Context) {
 	s.StopWaiter.Start(ctxIn, s)
 	backoff := time.Second
 	ephemeralErrorHandler := util.NewEphemeralErrorHandler(10*time.Minute, "is ahead of on-chain nonce", 0)
+
```

Comment:
The current way dual-protocol is handled is a little difficult to really understand, and hard to see exactly what's going on when bold is/isn't/becomes active.
I think best way would be something like:
* keep l1Validator and Staker as-is with as little as possible modification (possibly name change)
* form a separate "multiprotocol staker" or "switcher" or something similar (not sure how to name it)
* the switcher will hold Staker, BoldManager, and other fields it needs to choose between the two (possibly also a lock)
* the switcher will choose which protocol to use on Init, and if not bold will keep polling much like current implementation
* the switcher should have clear state about which protocol is used (e.g. either bold or legacy will always be nil)
* things both protocols need to support should be extracted to switcher - for example both legacy staker and bold manager implement "getStakedInfo", and the switcher calls the appropriate one
* arbnode will create/init/start the switcher, switcher will create/init/start as appropriate either legacy or bold. It'll probably create staker or bold only in Initialize step, because in that step there is a context and you can check status of parent chain

Reply:
Thanks @tsahee for the initial pass - I will mark back as a draft as I did a lot of hacky things today trying to get things to work on the testnet with the latest changes. Agreed we need to really improve how the switch happens in a cleaner way